### PR TITLE
IGNITE-23205 Fix ScaleCubeTopologyService.getById performance

### PR DESCRIPTION
**Pull request by ptupitsyn**
Replace stream-based brute force node lookup with a map in `ScaleCubeTopologyService.getById`.

### CODE REVIEW

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public class AbstractMultiNodeBenchmark {
     protected static Ignite publicIgnite;
     protected static IgniteImpl igniteImpl;
 
-    @Param({"false", "true"})
+    @Param({"false"})
```
Comment (ibessonov) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java:74]: Please return it back :)

**Review by ibessonov — APPROVED** [review]

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ */
+@State(Scope.Benchmark)
+@Fork(1)
+@Threads(1)
+@Warmup(iterations = 3, time = 2)
+@Measurement(iterations = 3, time = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class TableGetBenchmark extends AbstractMultiNodeBenchmark {
```
Comment (korlov42) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java:53]: we have `SelectBenchmark` which covers getByKey scenarios from all possible clients

Comment (ptupitsyn) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java]: Removed the new one.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    public static void main(String[] args) throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(".*" + TableGetBenchmark.class.getSimpleName() + ".embeddedGet*")
+                .build();
+
+        new Runner(opt).run();
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java]: Benchmark removed.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java (hunk 1 above)
Comment (ptupitsyn) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/AbstractMultiNodeBenchmark.java]: Unnecessary changes rolled back, `idToMemberMap` update logic fixed.

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java (hunk 2 above)
Comment (ascherbakoff) [reply modules/runner/src/integrationTest/java/org/apache/ignite/internal/benchmark/TableGetBenchmark.java]: I suggest to rename to KvGetBenchmark (we already have KvUpsertBenchmark)

**Review by ibessonov — APPROVED** [review]

File: modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TransactionInflights.java [hunk 4]
```diff
@@ -139,15 +139,8 @@ void cancelWaitingInflights(TablePartitionId groupId) {
...
-            if (ctx == null) {
-                ctx = new ReadOnlyTxContext();
-            }
-
-            ctx.finishTx(null);
-
-            return ctx;
-        });
+        var ctx = txCtxMap.computeIfAbsent(txId, k -> new ReadOnlyTxContext());
+        ctx.finishTx(null);
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TransactionInflights.java]: Ok, reverted.

**Pull request merged**